api/src/main/java/org/openmrs/api/db/hibernate/HibernateFormDAO.java

```diff
@@ -463,10 +463,10 @@ private Criteria getFormCriteria(String partialName, Boolean published, Collecti
 			DetachedCriteria subquery = DetachedCriteria.forClass(FormField.class, "ff");
 			subquery.setProjection(Projections.property("ff.form.formId"));
 			subquery.add(Restrictions.in("ff.formFieldId", anyFormFieldIds));
```

Comment:
Line 466 below is what am talking about

Reply:
Quiet interesting, that obviously doesn't take effect. 
1: My changes basically are a fix for filtration with argument <code>containingAllFormFields</code> of which where you commented is in a different criteria scope.
2: Have a second glance at my tests, you will find that even that line of code you reverted is not executed. Just look into this @dkayiwa

Reply:
Mr Kayiwa, Like you Hibernate Criteria. I had to change the line below
<code>Criteria crit = sessionFactory.getCurrentSession().createCriteria(Form.class, "form");</code> to <code>Criteria crit = sessionFactory.getCurrentSession().createCriteria(Form.class, "f");</code> so all references of 'form' needed to be updated abt the new 'f' alliance.

Basically the logic in the fix is this, The FormField Hibernate domain(Table) has a field <code>form</code>. Since the parent Criteria is owned by Form Hibernate domain/table, if we use an alliance of the same name as a field contained in the FormField domain(Detached  Criteria) Hibernate will get confused and do some thing like in the sql String below.

<code>select *  from form this_ where this_.published=? and ? = (select count(ff_.form_field_id) as y0_ from form_field ff_ where ff_.form_id = ff_.form_id and ff_.form_field_id in (?, ?, ?, ?)) </code>

instead of :
<code>select *  from form this_ where this_.published=? and ? = (select count(ff_.form_field_id) as y0_ from form_field ff_ where ff_.form_id = this_.form_id and ff_.form_field_id in (?, ?, ?, ?)) </code>

So just need another name/alliance apart from 'form'

Reply:
Am not asking why you changed it. All am saying is that if my reverting that line does not result into any test failure, then we have nothing to catch the bug, just in case someone changed it accidentally. Have you ever read about TDD?

Reply:
The thing is that you changed the wrong place. Then abt TDD, I we have the failing test in place now.

Reply:
Are you commits squashed into one?

Reply:
Nope, I had reproduced the issue, let me fix stuff